tests/test_stack.py

```diff
@@ -558,6 +558,20 @@ def test_describe_outputs_handles_stack_with_no_outputs(
         response = self.stack.describe_outputs()
         assert response == []
 
+    def test_diff_stack_sends_correct_request(self):
+        self.stack._template = Mock(spec=Template)
+        self.stack._template.body = "local_template_content"
+
+        self.stack.connection_manager.call.return_value = { "TemplateBody": "remote_template_content"}
+        response = self.stack.diff()
+        self.stack.connection_manager.call.assert_called_with(
+            service="cloudformation",
+            command="get_template",
+            kwargs={"StackName": sentinel.external_name}
+        )
+        print response
+        assert response == "--- remote\n+++ local\n@@ -1 +1 @@\n-remote_template_content\n+local_template_content\n"
```

Comment:
I'd like to see more test cases. Adding a line before a given line, adding a line after a given line, adding a line in between two lines, as well as sanity checks when you switch which template you consider "remote" and which template you consider "local", the same lines still show up.

An example of what I mean by the sanity check:

Consider the following files:
foo with contents:
```
bar
```
bar with contents:
```
bar
baz
```
doing ```diff foo bar``` (the Unix diff, not a Sceptre diff) shows that baz was added. doing ```diff bar foo``` shows baz being removed. The Sceptre diff should behave similarly.